packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	RichText,
+	useBlockProps,
+	useInnerBlocksProps,
+	BlockControls,
+} from '@wordpress/block-editor';
+import { ToolbarButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import HeadingLevelDropdown from './heading-level-dropdown';
+
+const DETAILS = [
+	[
+		'core/paragraph',
+		{
+			placeholder: __(
+				'Add text or blocks that will display when the details block is opened.'
+			),
+		},
+	],
+];
+
+function DetailsBlock( { attributes, setAttributes } ) {
+	const { level, summary, showContent } = attributes;
+	const tagName = 'h' + level;
+	const blockProps = useBlockProps();
+	const innerBlocksProps = useInnerBlocksProps(
+		{
+			className: 'wp-block-details__content',
+		},
+		{
+			template: DETAILS,
```

Comment:
Thinking about `allowedBlocks` here ([context](https://github.com/WordPress/gutenberg/issues/44539#issuecomment-1263238237))... I've been testing with the latest posts blocks, image, group and a bunch of others, but I think it might be useful to restrict such blocks as cover and media text etc. 

Maybe even restricting container blocks such as group and column to avoid any layout weirdness.

What do you reckon about allowing all text blocks, image block, audio, and video?

Reply:
Well, it is easier to start with a limitation, than trying to add one later.
Text and image blocks, yes. 
I am not as sure about the audio and video. If the block is used as a transcript for video it probably would be a bad idea for the block to include another video.

Reply:
I wonder if there is a way to have a disallow list rather than allowed list.

Reply:
> I wonder if there is a way to have a disallow list rather than allowed list.

I was wondering the same thing when looking at this! I'll make a note. It might be a useful feature.

Reply:
Some related discussion about allowed/disallowed on the original issue https://github.com/WordPress/gutenberg/issues/5540#issuecomment-386192518